File: hexa-cli/assets/scaffold/go/internal/usecases/increment.go
// Package usecases holds the application's verbs.
//
// Rule 3: usecases imports domain and ports only. It takes the port as a
// parameter and never chooses which adapter fills it; that choice belongs to
// the composition root alone.
package usecases

import (
	"{{name}}/internal/domain"
	"{{name}}/internal/ports"
)

// Increment advances the count by one and returns the new value.
func Increment(store ports.CounterStore) domain.Count {
	next := store.Load().Next()
	store.Save(next)
	return next
}

File: hexa-cli/assets/scaffold/ts/src/core/usecases/increment.ts
/**
 * Use cases — the application's verbs.
 *
 * Rule 3: `usecases/` imports `domain/` and `ports/` only. It takes the port
 * as a parameter and never chooses which adapter fills it; that choice belongs
 * to the composition root alone.
 */
import type { Count } from '../domain/count.js';
import type { CounterStore } from '../ports/counter-store.js';

/** Advance the count by one and return the new value. */
export function increment(store: CounterStore): Count {
  const next = store.load().next();
  store.save(next);
  return next;
}
